core/dbt/task/debug.py

```diff
@@ -95,81 +115,140 @@ def path_info(self):
         open_cmd = dbt.clients.system.open_dir_cmd()
         fire_event(OpenCommand(open_cmd=open_cmd, profiles_dir=self.profiles_dir))
 
-    def run(self):
+    def run(self) -> bool:
         if self.args.config_dir:
             self.path_info()
-            return not self.any_failure
+            return DEBUG_RUN_STATUS["pass"]
 
-        version = get_installed_version().to_version_string(skip_matcher=True)
+        version: str = get_installed_version().to_version_string(skip_matcher=True)
         fire_event(DebugCmdOut(msg="dbt version: {}".format(version)))
         fire_event(DebugCmdOut(msg="python version: {}".format(sys.version.split()[0])))
         fire_event(DebugCmdOut(msg="python path: {}".format(sys.executable)))
         fire_event(DebugCmdOut(msg="os info: {}".format(platform.platform())))
+
+        # Load profile if possible, then load adapter info (which requires the profile)
+        load_profile_status: SubtaskStatus = self._load_profile()
+        fire_event(DebugCmdOut(msg="Using profiles dir at {}".format(self.profiles_dir)))
         fire_event(DebugCmdOut(msg="Using profiles.yml file at {}".format(self.profile_path)))
         fire_event(DebugCmdOut(msg="Using dbt_project.yml file at {}".format(self.project_path)))
-        self.test_configuration()
-        self.test_dependencies()
-        self.test_connection()
+        if load_profile_status.run_status == RunStatus.PASS:
+            if self.profile is None:
```

Comment:
I just flipped per Colin because of the request to avoid `not None` and put the positive case first. I think his _is_ more inline with our inhouse style....I think at least 😅